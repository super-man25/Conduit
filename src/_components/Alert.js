// @flow
import * as React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import styled from 'styled-components';

import { colors, shadows, zIndexes } from '_constants';
import { actions } from '_state/alert';
import { Icon } from './Icon';
import { Flex } from './Flex';

const StyledApiAlert: React.ComponentType<{}> = styled(Flex)`
  position: absolute;
  right: 25px;
  top: 25px;
  z-index: ${zIndexes.BASE};
  width: 350px;
  background-color: ${colors.white};
  border-color: ${(props) =>
    props.type === 'apiError' ? colors.red : colors.green};
  box-shadow: ${shadows.SMALL};
  visibility: ${(props) => (props.show ? 'visible' : 'hidden')};
  cursor: pointer;
  padding: 25px;
`;

const StatusIcon: React.ComponentType<{}> = styled.div`
  margin-right: 15px;
`;

const StatusText: React.ComponentType<{}> = styled.div`
  font-size: 20px;
  font-weight: 300;
  letter-spacing: 1px;
  color: ${(props) => props.color};
  margin-bottom: 3px;
`;

const MsgText: React.ComponentType<{}> = styled.div`
  font-size: 13px;
  font-weight: 300;
`;

export const Alert = () => {
  const alertState = useSelector(({ alert }) => alert);
  const dispatch = useDispatch();
  const clearAlert = () => dispatch(actions.clear());

  const show = alertState.type !== null && alertState.message !== null;
  const statusColor =
    alertState.type === 'apiError' ? colors.red : colors.green;
  const statusText = alertState.type === 'apiError' ? 'Error:' : 'Success!';

  return (
    <StyledApiAlert {...alertState} show={show} onClick={clearAlert}>
      <StatusIcon>
        <Icon name={alertState.type} size={40} color={statusColor} />
      </StatusIcon>
      <Flex direction="column">
        <StatusText color={statusColor}>{statusText}</StatusText>
        <MsgText>{alertState.message}</MsgText>
      </Flex>
    </StyledApiAlert>
  );
};
