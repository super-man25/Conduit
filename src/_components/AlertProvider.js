import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { history } from '_helpers';
import { actions as alertActions } from '_state/alert';

class AlertProviderPresenter extends React.Component {
  constructor(props) {
    super(props);

    const { dispatch } = this.props;
    history.listen(() => {
      dispatch(alertActions.clear());
    });
  }

  render() {
    const { alert, children } = this.props;

    // TODO: style alerts
    return (
      <div>
        {alert &&
          alert.show && (
            <div className={`alert ${alert.type}`}>{alert.message}</div>
          )}
        {children}
      </div>
    );
  }
}

AlertProviderPresenter.propTypes = {
  /** Redux dispatch */
  dispatch: PropTypes.func,

  /** Alert object */
  alert: PropTypes.shape({
    message: PropTypes.string,
    type: PropTypes.string
  }),

  /** Children */
  children: PropTypes.node
};

function mapStateToProps(state) {
  return {
    alertState: state.alert
  };
}

export const AlertProvider = connect(mapStateToProps)(AlertProviderPresenter);
