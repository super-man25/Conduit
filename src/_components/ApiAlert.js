// @flow

import * as React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import { cssConstants, shadows, zIndexes } from '_constants';
import { Icon } from './Icon';

type Props = {
  alertState: {
    type: 'api-error' | 'api-success',
    message: string
  }
};

export const ApiAlertDiv: React.ComponentType<{}> = styled.div`
  position: absolute;
  right: 24px;
  top: 24px;
  z-index: ${zIndexes.BASE};
  font-size: 18px;
  width: 357px;
  height: 90px;
  background-color: ${cssConstants.PRIMARY_WHITE};
  border-color: ${(props) =>
    props.type === 'api-error'
      ? cssConstants.SECONDARY_RED
      : cssConstants.SECONDARY_GREEN};
  box-shadow: ${shadows.SMALL};
  visibility: ${(props) => (props.show ? 'visible' : 'hidden')};
`;

const StatusIcon: React.ComponentType<{}> = styled.div`
  position: absolute;
  top: 26px;
  left: 26px;
`;

const StatusText: React.ComponentType<{}> = styled.div`
  position: absolute;
  top: 26px;
  left: 93px;
  font-size: 1.2em;
  font-weight: 300;
  letter-spacing: 1px;
  color: ${(props) => props.color};
`;

const MsgText: React.ComponentType<{}> = styled.div`
  position: absolute;
  top: 54px;
  left: 93px;
  font-size: 0.7em;
  font-weight: 300;
  margin-right: 24px;
  color: ${cssConstants.PRIMARY_BLACK};
`;

export class ApiAlertPresenter extends React.Component<Props> {
  render() {
    const { alertState } = this.props;
    const show = {
      show: alertState.type !== null && alertState.message !== null
    };
    const statusColor =
      alertState.type === 'api-error'
        ? cssConstants.SECONDARY_RED
        : cssConstants.SECONDARY_GREEN;
    const statusText = alertState.type === 'api-error' ? 'Error:' : 'Success!';

    return (
      <ApiAlertDiv {...alertState} {...show}>
        <StatusIcon>
          <Icon name={alertState.type} size={39} color={statusColor} />
        </StatusIcon>
        <StatusText color={statusColor}>{statusText}</StatusText>
        <MsgText>{alertState.message}</MsgText>
      </ApiAlertDiv>
    );
  }
}

function mapStateToProps(state) {
  return {
    alertState: state.alert
  };
}

export const ApiAlert = connect(
  mapStateToProps,
  null
)(ApiAlertPresenter);
