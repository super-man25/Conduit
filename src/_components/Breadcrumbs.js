import styled from 'styled-components';
import { cssConstants } from '../_constants';

export const Breadcrumbs = styled.div`
  width: 100%;
  height: 30px;
  margin: 0;
  padding: 0;
  background: ${cssConstants.PRIMARY_WHITE};
  font-size: 12px;
`;
