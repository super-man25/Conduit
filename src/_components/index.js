export * from './Alert';
export * from './AsyncButton';
export * from './Breadcrumbs';
export * from './Button';
export * from './Center';
export * from './CenteredLoader';
export * from './ChartContainer';
export * from './ChartLegend';
export * from './Checkbox';
export * from './ChipButton';
export * from './ContentWrapper';
export * from './DatePicker';
export * from './DateRangeDropdown';
export * from './DateRangeFilter';
export * from './Dropdown';
export * from './DropdownButton';
export * from './EDLink';
export * from './EDText';
export * from './Flex';
export * from './Form';
export * from './FormInput';
export * from './FormInputArray';
export * from './HelpBlockDiv';
export * from './HorizontalBars';
export * from './Icon';
export * from './Icon/SVGIcons';
export * from './ImageLayout';
export * from './Input';
export * from './Integration';
export * from './Layout';
export * from './LeftNav';
export * from './LegendItem';
export * from './Loader';
export * from './MailtoLink';
export * from './MainContent';
export * from './Modal';
export * from './Panel';
export * from './ReportDownloadButton';
export * from './ScreenTitleBlock';
export * from './ScrollableList';
export * from './SecuredRoute';
export * from './SelectBox';
export * from './SelectDropdown';
export * from './Setting';
export * from './SettingButton';
export * from './SiteHeader';
export * from './SortableButton';
export * from './Spacing';
export * from './StyledTags';
export * from './Table';
export * from './Toggle';
export * from './UserWelcome';
