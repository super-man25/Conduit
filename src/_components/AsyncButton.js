// @flow

import * as React from 'react';
import { cssConstants } from '_constants';
import { PrimaryButton, Loader } from '_components';

type Props = {
  isLoading: boolean,
  disabled?: boolean,
  children: React.Node,
};

export const AsyncButton = (props: Props) => {
  const { isLoading, children, ...rest } = props;
  return (
    <PrimaryButton {...rest}>
      {props.isLoading ? (
        <Loader small color={cssConstants.PRIMARY_WHITE} />
      ) : (
        props.children
      )}
    </PrimaryButton>
  );
};
