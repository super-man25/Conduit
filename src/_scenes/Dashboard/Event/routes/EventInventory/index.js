// @flow
import React from 'react';
import { PageWrapper, Spacing, Box, Flex, FlexItem } from '_components';
import { connect } from 'react-redux';
import EventHeader from '../../components/EventHeader';
import type { EDEvent } from '_models';
import { selectors } from '_state/event';
import { VirtualizedEventInventory } from './components/InventoryTable/VirtualizedInventoryTable';

type Props = {
  event: ?EDEvent
};

export const EventInventory = ({ event }: Props) =>
  !!event && (
    <PageWrapper>
      <Flex direction="column" height="100%">
        <Box padding="1.5rem 2rem">
          <EventHeader availableInventory={15000} totalInventory={40000} />
        </Box>
        <Spacing height="2rem" />
        <FlexItem>
          <VirtualizedEventInventory event={event} />
        </FlexItem>
      </Flex>
    </PageWrapper>
  );

function mapStateToProps(state) {
  return {
    event: selectors.selectEvent(state)
  };
}

export default connect(mapStateToProps)(EventInventory);
