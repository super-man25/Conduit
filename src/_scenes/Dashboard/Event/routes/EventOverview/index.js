// @flow
import React from 'react';
import { PageWrapper, Spacing, Box } from '_components';
import { connect } from 'react-redux';
import EventHeader from '../../components/EventHeader';
import EventChart from './components/EventChart';
import type { EDEvent } from '_models';
import { selectors } from '_state/event';
import { TicketIntegrations } from './components/TicketIntegrations';
import { EventPricing } from './components/EventPricing';

type Props = {
  event: ?EDEvent,
  isAdmin: boolean
};

export const EventOverview = ({ event, isAdmin }: Props) =>
  !!event && (
    <PageWrapper>
      <Box padding="1.5rem 2rem">
        <EventHeader availableInventory={15000} totalInventory={40000} />
        <Spacing height="1rem" />
        <EventChart event={event} />
        <Spacing height="2rem" />
        <TicketIntegrations />
        <Spacing height="2rem" />
        {isAdmin && <EventPricing event={event} isAdmin={isAdmin} />}
      </Box>
    </PageWrapper>
  );

function mapStateToProps(state) {
  return {
    event: selectors.selectEvent(state),
    isAdmin: state.auth.model.isAdmin
  };
}

export default connect(mapStateToProps)(EventOverview);
