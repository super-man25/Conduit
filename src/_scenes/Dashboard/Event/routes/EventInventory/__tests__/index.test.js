import React from 'react';
import { shallow } from 'enzyme';
import { EventInventory } from '_scenes/Dashboard/Event/routes/EventInventory';
import { TertiaryButton } from '_components';

describe('<EventInventory /> index route', () => {
  const props = {
    event: {},
    selectedEventIds: [],
    isBulkUpdating: false,
    startBulkUpdate: jest.fn()
  };

  it('should render correctly', () => {
    const wrapper = shallow(<EventInventory {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when there are selectedEventIds', () => {
    const wrapper = shallow(
      <EventInventory {...props} selectedEventIds={[1, 2, 3]} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when there are selectedEventIds and isBulkUpdating is true', () => {
    const wrapper = shallow(
      <EventInventory
        {...props}
        selectedEventIds={[1, 2, 3]}
        isBulkUpdating={true}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should trigger startBulkUpdate when the bulk update button is clied', () => {
    const fn = jest.fn();
    const wrapper = shallow(
      <EventInventory
        {...props}
        selectedEventIds={[1, 2, 3]}
        isBulkUpdating={true}
        startBulkUpdate={fn}
      />
    );

    wrapper
      .find(TertiaryButton)
      .at(0)
      .simulate('click');

    expect(fn).toBeCalled();
  });
});
