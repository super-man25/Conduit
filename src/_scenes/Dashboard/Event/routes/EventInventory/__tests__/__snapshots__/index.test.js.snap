// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventInventory /> index route should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    event={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Flex
        direction="column"
        height="100%"
>
        <styled.div
                padding="1.5rem 2rem"
        >
                <Connect(withSidebarEventHeader)
                        availableInventory={15000}
                        totalInventory={40000}
                />
        </styled.div>
        <styled.div
                height="2rem"
        />
        <FlexItem
                flex={1}
        >
                <Connect(VirtualizedEventInventoryPresenter)
                        event={Object {}}
                />
        </FlexItem>
</Flex>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div
            padding="1.5rem 2rem"
>
            <Connect(withSidebarEventHeader)
                        availableInventory={15000}
                        totalInventory={40000}
            />
</styled.div>,
          <styled.div
            height="2rem"
/>,
          <FlexItem
            flex={1}
>
            <Connect(VirtualizedEventInventoryPresenter)
                        event={Object {}}
            />
</FlexItem>,
        ],
        "direction": "column",
        "height": "100%",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(withSidebarEventHeader)
              availableInventory={15000}
              totalInventory={40000}
/>,
            "padding": "1.5rem 2rem",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "availableInventory": 15000,
              "totalInventory": 40000,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "height": "2rem",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(VirtualizedEventInventoryPresenter)
              event={Object {}}
/>,
            "flex": 1,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "event": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Flex
          direction="column"
          height="100%"
>
          <styled.div
                    padding="1.5rem 2rem"
          >
                    <Connect(withSidebarEventHeader)
                              availableInventory={15000}
                              totalInventory={40000}
                    />
          </styled.div>
          <styled.div
                    height="2rem"
          />
          <FlexItem
                    flex={1}
          >
                    <Connect(VirtualizedEventInventoryPresenter)
                              event={Object {}}
                    />
          </FlexItem>
</Flex>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.div
              padding="1.5rem 2rem"
>
              <Connect(withSidebarEventHeader)
                            availableInventory={15000}
                            totalInventory={40000}
              />
</styled.div>,
            <styled.div
              height="2rem"
/>,
            <FlexItem
              flex={1}
>
              <Connect(VirtualizedEventInventoryPresenter)
                            event={Object {}}
              />
</FlexItem>,
          ],
          "direction": "column",
          "height": "100%",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Connect(withSidebarEventHeader)
                availableInventory={15000}
                totalInventory={40000}
/>,
              "padding": "1.5rem 2rem",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "availableInventory": 15000,
                "totalInventory": 40000,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "height": "2rem",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Connect(VirtualizedEventInventoryPresenter)
                event={Object {}}
/>,
              "flex": 1,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "event": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
